BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb

```diff
@@ -318,25 +340,89 @@
   <%- end -%>
   <%- if param['description'] -%>
         # Description
-        description.(<<-'DESC')
+        description.(<<-'DESC',<%= isGroupMember %>)
           <%= param['description'] %>
         DESC
   <%- end -%>
-      %>
+     %> 
     </li>
-
+<%- end -%>
+<%# Generate HTML for a group -%>
+<%- writeGroup = lambda do |group| -%>
+  <%-
+    # Get group parameters 
+    id = group['id']
+    name = group['name']
+    groupDescription = group['description']
+    groupParams = gIdToPrms[id]
+    mbrs = gIdToMbrs[id]
+    defaultChecked = group['one-is-required'] || required.any? { |p| mbrs.include? p["id"] }  
+  -%>
+  <div class="group <%= id %>"> 
+    <%# Write group header %> 
+    <div style="border-bottom: 2px solid #bbbbbb; padding-bottom: 2px;">
+      <h3 style="border-width: 0px; margin: 0px; padding: 1px; padding-left: 0px;">
+        <%= name %> <%% groupCheckbox.('<%= id %>',<%= defaultChecked %>) %>  
+      </h3>
+      <h5 style="font-size : 9.5pt; font-weight: normal; padding-top: 2px;" class="grp-desc"> 
+        <%= groupDescription %> 
+      </h5> 
+    </div><%%= "<br>".html_safe %>
+    <%# Write group parameters (required ones first) %>
+    <%- groupParams.each { |p| parameter.(p, true) if required.any? { |r| p["id"]==r["id"] } } -%>
+    <%- groupParams.each { |p| parameter.(p, true) if optional.any? { |t| p["id"]==t["id"] } } -%>
+  </div>
+  
 <%- end -%>
 <div class="task-params">
   <%- if params.empty? -%>
   <%= name %> has no parameters.
   <%- else -%>
   <ul>
-    <%- required.each(&parameter) -%>
-    <%- optional.each(&parameter) -%>
+%   # Write info for ungrouped parameters
+    <%- for param in required do parameter.(param) if whichGroup.(param["id"]).nil? end -%>
+    <%- for param in optional do parameter.(param) if whichGroup.(param["id"]).nil? end -%>
+%   # Write info for grouped parameters
+    <%- groups.each(&writeGroup) unless noGroups  -%>
   </ul>
   <%- end -%>
 </div>
 
+%# Style block for group toggling checkbox
```

Comment:
Should style be embedded or put in a separate file in the CBRAIN framework?

e.g. `BrainPortal/public/stylesheets/boutique.css`